ruby-gem/lib/calabash-android/drag_helpers.rb

```
@@ -0,0 +1,109 @@
+require 'calabash-android/env'
+
+module Calabash
+  module Android
+    module DragHelpers
+
+      MAX_RETRIES = 7
+      @@monkey_port = nil
+
+      def start_monkey
+        kill_existing_monkey_processes
+        retries = 0
+        loop do
+          port = rand((1025..65535))
+          output = ''
+          fork do
+            output = `#{Env.adb_path} shell monkey --port #{port}`
+          end
+          sleep(2.0)
+          unless output.include? 'Error binding to network socket.'
+            #The shim will always exit, but on success it will not output the line above
+            pids = existing_monkey_pids
+            if pids.count == 1 then
+              return port
+            end
+          end
+          break if (retries += 1) > MAX_RETRIES
```

We should raise an error if we are unable to start monkey